flink-table-store-core/src/main/java/org/apache/flink/table/store/file/mergetree/compact/AggregationMergeFunction.java

```diff
@@ -0,0 +1,150 @@
+        this.aggregateColumnNames = aggregateColumnNames;
+
+        List<LogicalType> fieldTypes = rowType.getChildren();
+        this.getters = new RowData.FieldGetter[fieldTypes.size()];
+        for (int i = 0; i < fieldTypes.size(); i++) {
+            getters[i] = RowData.createFieldGetter(fieldTypes.get(i), i);
+        }
+
+        this.isPrimaryKey = new boolean[this.getters.length];
+        Arrays.fill(isPrimaryKey, false);
+        List<String> rowNames = rowType.getFieldNames();
+        for (String primaryKeyName : primaryKeyType.getFieldNames()) {
+            isPrimaryKey[rowNames.indexOf(primaryKeyName)] = true;
+        }
+
+        this.aggregateFunctions = new ArrayList<>(rowType.getFieldCount());
+        for (int i = 0; i < rowType.getFieldCount(); i++) {
+            AggregateFunction<?> f = null;
+            if (aggregateColumnNames.contains(rowNames.get(i))) {
+                f = choiceRightAggregateFunction(rowType.getTypeAt(i).getDefaultConversion());
+            }
+            aggregateFunctions.add(f);
+        }
+    }
+
+    private AggregateFunction<?> choiceRightAggregateFunction(Class<?> c) {
+        AggregateFunction<?> f = null;
+        if (Double.class.equals(c)) {
+            f = new DoubleAggregateFunction();
+        } else if (Long.class.equals(c)) {
+            f = new LongAggregateFunction();
+        } else if (Integer.class.equals(c)) {
+            f = new IntegerAggregateFunction();
+        } else if (Float.class.equals(c)) {
+            f = new FloatAggregateFunction();
+        }
+        return f;
+    }
+
+    @Override
+    public void reset() {
+        this.row = new GenericRowData(getters.length);
+    }
+
+    @Override
+    public void add(RowData value) {
+        for (int i = 0; i < getters.length; i++) {
+            Object currentField = getters[i].getFieldOrNull(value);
+            AggregateFunction<?> f = aggregateFunctions.get(i);
+            if (isPrimaryKey[i]) {
+                // primary key
+                if (currentField != null) {
+                    row.setField(i, currentField);
+                }
+            } else {
+                if (f != null) {
+                    f.reset();
+                    Object oldValue = row.getField(i);
+                    if (oldValue != null) {
+                        f.aggregate(oldValue);
+                    }
+                    switch (row.getRowKind()) {
+                        case INSERT:
+                            f.aggregate(currentField);
+                            break;
+                        case DELETE:
+                            f.aggregate(currentField, false);
+                            break;
+                        case UPDATE_AFTER:
+                        case UPDATE_BEFORE:
+                        default:
+                            throw new UnsupportedOperationException(
+                                    "Unsupported row kind: " + row.getRowKind());
```

Comment:
You can treat `UPDATE_BEFORE` as `DELETE` and `UPDATE_AFTER` as `INSERT`. They're actually different in some other ways but in our scenario they're the same.